### DISCUSSION TITLE
Allow negative habits

### DISCUSSION THREAD

**Discussion by a-r-m-i-n** [iSoron/uhabits#1311] [Feature Requests] [answered]
Currently you can only tell a miminum required to reach the goal. It would be also nice to have habits like "smoke 5 or less cigarettes a day" or "drink alcohol less than twice a week".

To achieve this you would only need to add an additional checkbox, right after the current goal settings, to "invert the goals".

Btw. thanks for this great app!

**Comment by LordJon** [comment]
You can already kinda do this right? I have habit to use my smartphone less than 3 hours a day. It doesn't count the days (I'm still on version 1.x that hasn't that option) it's just an boolean that I can check.

From my understanding the developers have the intention to motivate to use positive habits instead of having negative ones.
I think yours will fall in between as the goal still is positive.

I guess if you want to change this habit over time by changing the maximum amount of cigarettes the stats will be recalculated and then this feature actually makes sense. Could be interesting.

**Reply by a-r-m-i-n** [reply]
My feature request is about numeric habits. Of course you could turn everything to a boolean "did you make it?", but then you loose all statistics. And I would like to count the actual amount of cigarettes i.e. with keeping all stats. Thanks to that, I know, that I'm smoking less on Tuesdays ;-) 

But the app tells me, that my goal is reached when I smoke more. Which is the opposite, of what the developer maybe intially intended.

**Reply by TheSprinklizer** [reply]
+1 - I would like to see the same.
How many cigarettes? Goal <6
How many calories? Goal <1200
Rephrasing doesn't work for 2 reasons:
1- I lose the ability to track the number, and
2- at the end of the day, I may not remember everything. I need to enter the number as it happens, or I may forget a cigarette or a snack. 
As others have mentioned, it's not about a negative habit or addiction. It's about setting a goal that's a maximum instead of a minimum. 
I have tried MANY habit tracking apps, and this is my favorite by far - the best historical reporting I've seen. If it had this option for a maximum goal, it would be perfect.

**Comment by ppseprus** [comment]
I'd like to pile on the above discussion! Only recently have I started using this app, and I have already sent an email to the creator — not realising there's a Github link there 😅 

Instead of "negative" habits, I think we should simply allow the option of setting whether the target of your measurable habit is a **min** or a **max**.

Here are some scenarios that come to mind: Any over-consumption scenarios: caffeine, alcohol, cigarettes, eating out or ordering instead of cooking, time spent under the shower etc.
* Eg. you'd like to reduce your caffeine intake (clearly, a positive habit), you're already having 6-7 espressos a day and you want to set a max target of 4
* Eg. you'd like to reduce the amount of time you spend under the shower: you're spending 15 minutes, while 8 minutes would be better for the environment and your skin

I think these are still about building and promoting good (or at least better) habits, but the current version of the application do not really support tracking them.

**Reply by GollyTicker** [reply]
I totally support the idea if such min max goals. I track my weight - but I cannot tell the app - that my goal is to be lower than the target. Currently, when I put X kg as the target and my weight is actually X +1 kg, then the app registers it as a successful day 😅

I really like the app - and could also give some support here if wished. I would like to be able to define my target weight as the desired  upper bound. 

I tried a trick with using negative numbers - but negative numbers cannot be used as values or targets...

**Reply by a-r-m-i-n** [reply]
For weight tracking I can stronly recommend the app [link to non-open-source app removed]

**Reply by iSoron** [reply]
@a-r-m-i-n App recommendations are welcome, as long as they are open source. Recommendations to closed source apps will be removed.

**Reply by a-r-m-i-n** [reply]
Oh, I actually thought it is open source, but you are right.
However, I think weight tracking is not the purpose of your app.

**Reply by ppseprus** [reply]
This thread is diverging a bit I think. The main point here is that there are a ton of scenarios where a min/max checkbox feature would be more than useful, and would actually extend the use cases of this application beyond its current capabilities.

**Comment by Silviopina** [comment]
Hi folks, I love the Lopp habits app, I have recommended it  and loved following the discussion here for a bit, but I have desperately, yet unsuccessfully been trying to unsubscribe and don't know what I am doing wrong. I have clicked on the"unsubscribe" button at the bottom of the emails, which takes me here and I can read the words "You’ve been unsubscribed from this thread." at the top, but I keep getting emails nonetheless. Please help!
Thank you! I would prefer to follow from Github, but not subscribe, thanks!
Silvia

**Reply by abel-baker** [reply]
You may need to ignore/unwatch this repo as a whole rather than unsubscribing from individual threads.

**Comment by vsemecky** [comment]
At the top right, press `Unwatch` and select `Ignore`.
![obrazek](https://user-images.githubusercontent.com/238600/166952670-7cccdc67-89f0-4112-b6e3-fc70ee6bbb7b.png)

**Comment by drawtheliner** [comment]
+1 To quote @ppseprus 

> This thread is diverging a bit I think. The main point here is that there are a ton of scenarios where a min/max checkbox feature would be more than useful, and would actually extend the use cases of this application beyond its current capabilities.

**Comment by exanrauzer** [comment]
Yes, this thread by it's wide topic 'Allow negative habbits' is now a little dead and maybe should be discarded to unveil the grip for fresh perspective.

In my vision negative habbits starts with 100% - you admit that you are addicted and you checkmark when you DO the activity that you are addicted to. This is the redefining the paradigm of addiction-help apps that are pointless when you checkmark the days that you are clear - it causes that you do not use an app if you are into some several days run with your addiction. So you drop the tool that needs to help you to get away of the addiciton when you do the addiction??

And other points that this aproach is flawed, not mentioned now.

When you checkmark an addiction day instead - you stay into the accountability zone - you got the full information of your negative streak, your run. You can't fool yourself, that you got a day break with your bad habbit if the app tells you that is clearly not.

I would be extremly happy to have this feature here in loop. Simple changes:

- Mark habbit as 'negative' in habbit creating menu (that gives you 100% start)
- Checkmarks are nverted:
  - the day with an addiction is a colorful cross (red I suggest)
  - the clear day (without a addiction) is hollow checmark.
- Reverse ring chart moving direction in main screen or replace with a full dot (not ring) that becomes smaller, and smaller till the dissappear.

I whink I will start a new discussion with this proposal.

Thank you for reading.

**Comment by ppseprus** [comment]
@exanrauzer, I agree that this app should not be used for managing addiction, but I also think that between what is currently supported by the app and _addiction-management_, you're just missing a range of use cases.

First of all, not every instance where you consume more than you want to should be considered an _addiction_. Second, there are other activities besides _consumption_.

My own examples [above](https://github.com/iSoron/uhabits/discussions/1311#discussioncomment-2688909),  are about establishing more conscious habits.

I drink coffee every day, but in a varying amount. Your reverse-logic would not work because the 100% is not a fixed amount.

Showering less, while we're actually experiencing a severe drought, is again, not the correction of an addiction, it's not necessarily a bad habit, but for some reason you'd like to improve upon this activity in a direction that currently cannot be tracked by the app.

**Comment by ppseprus** [comment]
Seems like this feature has been shipped! 😎 🙇

**Reply by a-r-m-i-n** [reply]
Indeed. Thank you very much! ❤️

**Comment by a-r-m-i-n** [comment, marked as answer]
Feature has been shipped. Thank you very much
